test_conformance/extensions/cl_khr_command_buffer/command_buffer_test_kernel.cpp

```diff
@@ -0,0 +1,98 @@
+//
+// Copyright (c) 2022 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "basic_command_buffer.h"
+#include "harness/typeWrappers.h"
+#include "procs.h"
+
+#include <vector>
+
+
+namespace {
+
+////////////////////////////////////////////////////////////////////////////////
+// Command-queue fill tests which handles below cases:
+// 
+// - barrier wait list
+
+    struct BarrierWithWaitListKHR : public BasicCommandBufferTest
+    {   
+        
+        using BasicCommandBufferTest::BasicCommandBufferTest;
+
+        cl_int Run() override
+        {
+            cl_int error = clCommandNDRangeKernelKHR(
+                command_buffer, nullptr, nullptr, kernel, 1, nullptr, &num_elements,
+                nullptr, 0, nullptr, nullptr, nullptr);
+            test_error(error, "clCommandNDRangeKernelKHR failed");
+
+            error = clCommandBarrierWithWaitListKHR(command_buffer, nullptr, 0, nullptr, nullptr, nullptr);
+            test_error(error, "clCommandBarrierWithWaitListKHR failed");
```

Comment:
As far as I understand this function waits for sync-points or non-blocking operations to be completed, thus you should precede it with such calls which is missing at the moment

Reply:
Done